pilz_psenscan_test_area/automatic_acceptance_test_psen_scan.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+[[ -z $TEST_AREA_SCRIPTS_PATH ]] && { echo "Environment variable TEST_AREA_SCRIPTS_PATH is not set! Cannot continue!" >&2; exit 1; }
+[[ -z $1 ]] && export TARGET_BRANCH="melodic-devel" || export TARGET_BRANCH="$1"
+
+TMP_DIR="/tmp"
+CATKIN_WS_NAME="catkin_ws"
+export CATKIN_WS_DIR="$TMP_DIR/$CATKIN_WS_NAME"
+export SRC_DIR="$CATKIN_WS_DIR/src"
+export REPO_NAME="psen_scan"
+export REPO_DIR="$SRC_DIR/$REPO_NAME"
+
+
+exit_failure()
+{
+  echo -e "\033[1;31mError! $1" >&2
+  echo -e "\033[1;31mTest failed!" >&2
+  echo -e "\033[1;37mTo run the test again, please execute '$TEST_AREA_SCRIPTS_PATH/automatic_acceptance_test_psen_scan.sh'"
+  exit 1
+}
+
+create_log_file()
+{
+  LOG_DIR="/var/log/automatic_acceptance_test"
+  mkdir $LOG_DIR 2> /dev/null
+
+  LOG_FILE_NAME="$(date '+%Y_%m_%d_%H_%M_%S')_"$REPO_NAME"_automatic_acceptance_test_log.txt"
+  LOG_FILE="$LOG_DIR/$LOG_FILE_NAME"
+  touch $LOG_FILE 2> /dev/null || { LOG_DIR="/home/$USER"; LOG_FILE="$LOG_DIR/$LOG_FILE_NAME"; echo "Cannot create system logfile." >&2; }
+  touch $LOG_FILE 2> /dev/null || exit_failure "Cannot create local logfile."
+  echo "Logging to $LOG_FILE."
+}
+
+source_ROS()
+{
+  source /opt/ros/melodic/setup.bash 2>> $LOG_FILE || exit_failure "Sourcing ROS failed!"
+}
+
+print_separator()
+{
+  separator=""
+  for i in $(seq 20)
+  do
+    separator+=$1
+  done
+  echo $separator
+}
+
+log_file_section()
+{
+  print_separator $2 >> $LOG_FILE
+  echo $1 >> $LOG_FILE
+  print_separator $2 >> $LOG_FILE
+}
+
+start_log_file_section()
+{
+  log_file_section "$1" ">"
+}
+
+end_log_file_section()
+{
+  log_file_section "$1" "<"
+}
+
+execute_section()
+{
+  section_name=$(echo "$1" | sed -E 's/.*\/([a-zA-Z_]+)\.sh/\1/;s/_/ /g')
+  echo "$section_name ..."
+  start_log_file_section "$section_name"
+  $@ >> $LOG_FILE || exit_failure "$1 failed!"
```

Comment:
Do successful sections not finish with `exit 0`? If they do, the `exit_failure` function would be called. Therefore, I assume that the first condition needs to be inverted, to avoid the execution of the `exit_failure` method.

Reply:
The command after `||` operator is only executed after a non-zero exit.